robot/server/server.go

```diff
@@ -165,6 +173,10 @@ func (s *Server) DiscoverComponents(ctx context.Context, req *pb.DiscoverCompone
 			return nil, errors.Wrapf(err, "unable to construct a structpb.Struct from discovery for %q", discovery.Query)
 		}
 		pbQuery := &pb.DiscoveryQuery{Subtype: discovery.Query.API.String(), Model: discovery.Query.Model.String()}
+		if nonTriplet {
+			pbQuery.Subtype = string(discovery.Query.API.ResourceSubtype)
+			pbQuery.Model = string(discovery.Query.Model.Name)
+		}
```

Comment:
cool, this should preserve discovery functionality with what app is currently sending